packages/@uppy/transloadit/src/Assembly.js

```diff
@@ -62,6 +65,59 @@ class TransloaditAssembly extends Emitter {
     this.close()
   }
 
+  #connectServerEvent () {
+    this.#sse = new EventSource(this.status.assembly_ssl_url)
+
+    this.#sse.addEventListener('open', () => {
+      if (this.socket) {
+        this.socket.disconnect()
+        this.socket = null
+      }
+      clearInterval(this.pollInterval)
+      this.pollInterval = null
+    })
+
+    /*
+     * The event "message" is a special case, as it
+     * will capture events without an event field
+     * as well as events that have the specific type
+     * other event type.
+     */
+    this.#sse.addEventListener('message', (e) => {
+      if (e.data === 'assembly_finished') {
+        this.#onFinished()
+      }
+
+      if (e.data.startsWith('assembly_upload_finished:')) {
+        const file = JSON.parse(e.data.slice('assembly_upload_finished:'.length))
+        this.emit('upload', file)
+        this.status.uploads.push(file)
+      }
+
+      if (e.data === 'assembly_uploading_finished') {
+        this.emit('executing')
+      }
+
+      if (e.data === 'assembly_upload_meta_data_extracted') {
+        this.emit('metadata')
+        this.#fetchStatus({ diff: false })
+      }
+
+      if (e.data.startsWith('assembly_result_finished:')) {
+        const [stepName, result] = JSON.parse(e.data.slice('assembly_result_finished:'.length))
+        this.emit('result', stepName, result)
+        ;(this.status.results[stepName] ??= []).push(result)
```

Comment:
shall we pull out these two lines into a method called this.#onResultFinished ? as now it's duplicated between the sse and sockets